### PR TITLE
add reduceCopy solution

### PR DESCRIPTION
**Pull request by DidenkoDenis306**

### CODE REVIEW

File: src/arrayMethodReduce.js [hunk 1]
```diff
@@ -5,7 +5,14 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acc = arguments.length > 1 ? initialValue : this[0];
```
Comment (ghost) [inline src/arrayMethodReduce.js]: let's make our variables clearer and call it `accumulator` instead of `acc`

File: src/arrayMethodReduce.js [hunk 2]
```diff
@@ -5,7 +5,14 @@
...
+    const start = arguments.length > 1 ? 0 : 1;
```
Comment (ghost) [inline src/arrayMethodReduce.js]: do we need such complicated checks? we make the same thing in one `if` statement

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/arrayMethodReduce.js [hunk 3]
```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let accumulator = initialValue;
+    let start = 0;
```
Comment (dinaminahi) [inline src/arrayMethodReduce.js]: Can we come up with more descriptive name here? something like `startIndex`

**Review by dinaminahi — CHANGES_REQUESTED** [review]

**Review by Anna-Tkachenko — APPROVED** [review]
⭐️

**Pull request closed without merging**